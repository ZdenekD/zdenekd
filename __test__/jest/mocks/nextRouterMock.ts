const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        asPath: '/page/subpage/something-else',
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
