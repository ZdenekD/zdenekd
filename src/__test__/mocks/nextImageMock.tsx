const nextImageMock = jest.mock('next/image', () => function Image() {
    return <img src="" alt="" />;
});

export default nextImageMock;
