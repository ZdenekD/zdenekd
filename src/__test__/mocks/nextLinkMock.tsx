const nextLinkMock = jest.mock('next/link', () => function Link() {
    return <a href="/">link</a>;
});

export default nextLinkMock;
