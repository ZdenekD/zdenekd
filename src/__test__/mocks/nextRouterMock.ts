const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'en',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
