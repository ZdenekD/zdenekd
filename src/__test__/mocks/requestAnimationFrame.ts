import react from 'react';

window.requestAnimationFrame = () => 0;

export default react;
