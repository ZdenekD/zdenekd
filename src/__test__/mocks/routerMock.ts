const routerMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
    })),
}));

export default routerMock;
