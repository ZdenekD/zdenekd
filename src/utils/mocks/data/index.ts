import MockDataParagraph from './content/UI/paragraph';

export {MockDataParagraph};
