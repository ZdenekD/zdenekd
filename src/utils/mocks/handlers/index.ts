import handler from './handler';

const handlers = [...handler];

export default handlers;
