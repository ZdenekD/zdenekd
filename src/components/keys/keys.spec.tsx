import {render, screen} from '@testing-library/react';
import '@/test/mocks/nextRouterMock';
import Keys from '.';

describe('Form/Keys', () => {
    it('renders without error', () => {
        render(<Keys />);

        expect(screen.getByTestId('component-keys')).toBeInTheDocument();
    });
});
