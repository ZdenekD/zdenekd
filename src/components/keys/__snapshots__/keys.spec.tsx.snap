// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keys match snapshot 1`] = `
<div class="keys"
     data-test="component-keys"
>
  <button type="button"
          class="key "
          aria-label="Předchozí stránka"
          data-test="component-key"
  >
    <i class="icon up">
    </i>
  </button>
  <div class="key">
    <button type="button"
            class="key disabled"
            aria-label="Předchozí projekt"
            data-test="component-key"
            disabled
    >
      <i class="icon left">
      </i>
    </button>
    <button type="button"
            class="key "
            aria-label="Následující stránka"
            data-test="component-key"
    >
      <i class="icon down">
      </i>
    </button>
    <button type="button"
            class="key disabled"
            aria-label="Následující projekt"
            data-test="component-key"
            disabled
    >
      <i class="icon right">
      </i>
    </button>
  </div>
</div>
`;
